### PR TITLE
Unification of error messages and [SWARM-1881] Is possible to save item without a name

### PR DESCRIPTION
**Pull request by akoniar**
@Ladicek please review

### CODE REVIEW

**Comment by Ladicek** [comment]
ok to test

File: src/main/java/io/openshift/boosters/FruitResource.java [hunk 1]
```diff
@@ -82,8 +87,9 @@ public Response create(Fruit fruit) {
     @Produces("application/json")
     @Transactional
     public Response update(@PathParam("id") Integer id, Fruit fruit) {
-        if (fruit.getName() == null) {
-            return error(422, "Fruit Name was not set on request.");
+        
+        if (fruit == null || fruit.getName() == null || fruit.getName().trim().length() == 0) {
+            return error(400, "The name is required!");
```
Comment (Ladicek) [inline src/main/java/io/openshift/boosters/FruitResource.java]: I think 422 is more appropriate.

Comment (akoniar) [reply src/main/java/io/openshift/boosters/FruitResource.java]: Nodejs and Vert.x use 400, but for the e2e tests it isn't important to have same status code for this case so I will fix it

**Review by Ladicek — APPROVED** [review]

**Comment by kenfinnigan** [comment]
As per https://github.com/fabric8-launcher/launcher-documentation/wiki/Mission:-Relational-Database-Backend I think the response needs to be 415 when there's no payload and only 422 when it's invalid (ie. missing data)

**Comment by akoniar** [comment]
@kenfinnigan Thank for comment. It makes sense.

**Comment by Ladicek** [comment]
Based on the description of status code 415, I'd guess that would already be handled by JAX-RS (unsupported content type, or couldn't deserialize from JSON), but I didn't verify. All the extra validations we do here are on the `Fruit` object, hence deserialization was successful. But @akoniar please verify :-)

**Comment by akoniar** [comment]
@Ladicek If I set header for example xml header, then JAX-RS will handle it, but If is set JSON header and body is set as empty, then backend return 500.

**Comment by Ladicek** [comment]
OK, cool, thanks!

**Comment by Ladicek** [comment]
Could you please rebase on current `master`? Thanks.

**Comment by akoniar** [comment]
Done and I created a new pull request #33

**Pull request closed without merging**